### DISCUSSION TITLE
For generic jdbc driver connection, from which view does dbeaver retrieve index info?

### DISCUSSION THREAD

**Discussion by lightsailpro** [dbeaver/dbeaver#11609] [General]
I set up a connection with a generic jdbc driver. The "Indexes" folder is not populated on schema and table level. Click the Index folder shows nothing. Does anyone know for genric jdbc driver connection, from which standard system view dbeaver retrieve index info, or which source file contain the code? Thanks in advance.

**Comment by LonwoLonwo** [comment]
Hello @lightsailpro 
What is your driver/database?
for Generic drivers we are reading index info directly from driver with java.sql.DatabaseMetaData.getIndexInfo
and drivers usually read indexes from system views/tables.

**Reply by lightsailpro** [reply]
It is jdbc driver for splicemachine (https://splicemachine.com/), which is apache derby on hbase basically. I will see if I can find the exact system view it is querying. Thanks

**Reply by lightsailpro** [reply]
@LonwoLonwo: I am not very familiar with Java. The JDBCDatabaseMetaDataImpl.java does not seem to mention the exact system view name for the index. Is that index system view name info packaged in the jdbc jar itself, or hardcoded in the dbeaver source code? Thanks again.

**Reply by LonwoLonwo** [reply]
In the jdbc jar itself, yes.

https://doc.splicemachine.com/cmdlineref_showindexes.html - have you try SHOW INDEXES [ IN schemaName | FROM tableName ]?

**Comment by LonwoLonwo** [comment]
https://github.com/dbeaver/dbeaver/blob/e3181bf7b7acdbb5fa8f245a8ee4beacef183b72/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/exec/JDBCDatabaseMetaDataImpl.java#L1240 it's here in DBeaver project

**Comment by laurentgo** [comment]
@LonwoLonwo you might have mentioned me by mistake instead of @lightsailpro .

**Reply by LonwoLonwo** [reply]
oh! sorry
